bundle/jsky.app.ot/src/main/java/jsky/app/ot/gemini/editor/targetComponent/TelescopePosTableWidget.java

```diff
@@ -350,9 +311,172 @@ private static String extractName(final GuideGroup group) {
             return tmpRows;
         }
 
+        // Replace the auto group and return a new TableData.
+        // Note that we work under the assumption that there is ALWAYS an auto group, which should be the case,
+        // and that it is in row position 1 of the table (directly after the base).
+        void replaceAutoGroup(final Option<ObsContext> ctxOpt, final TargetEnvironment newEnv) {
+            final List<Row>         oldRows = rows;
+
+            // Recalculate the rows to see if quality has changed based on the new context.
+            env = newEnv;
+            final Set<Integer> changedRowIndices      = recalculateManualGSQuality(ctxOpt);
+
+            // The auto group is always in group position 0 and row position 1.
+            // Recreate it and set it in the new rows.
+            final GuideGroup autoGroup  = newEnv.getGuideEnvironment().getOptions().head();
+            final GroupRow autoGroupRow = createGroupRow(ctxOpt, 0, autoGroup);
+            rows.set(1, autoGroupRow);
+            numRows = countRows();
+
+            // Determine the size difference between the old auto group and the new.
+            final int oldAutoGroupSize  = oldRows.get(1).children().size();
+            final int newAutoGroupSize  = autoGroupRow.children().size();
+            final int autoGroupSizeDiff = newAutoGroupSize - oldAutoGroupSize;
+            final int autoGroupSizeInt  = Math.min(oldAutoGroupSize, newAutoGroupSize);
+
+            // Fire events indicating the changes in the model.
+            if (autoGroupSizeInt > 0) {
+                fireTableRowsUpdated(FIRST_AUTO_TARGET_IDX, FIRST_AUTO_TARGET_IDX + autoGroupSizeInt - 1);
+            }
+            if (oldAutoGroupSize < newAutoGroupSize) {
+                fireTableRowsInserted(FIRST_AUTO_TARGET_IDX + oldAutoGroupSize, FIRST_AUTO_TARGET_IDX + newAutoGroupSize - 1);
+            } else if (newAutoGroupSize < oldAutoGroupSize) {
+                fireTableRowsDeleted(FIRST_AUTO_TARGET_IDX + newAutoGroupSize, FIRST_AUTO_TARGET_IDX + oldAutoGroupSize - 1);
+            }
+
+            // Now for each group that was changed, fire update events.
+            int tableRowIdx = 0;
+            final Stream<Tuple2<Integer,Row>> indexedRows = IntStream.range(0, rows.size()).mapToObj(i -> new Pair<>(i, rows.get(i)));
+            for (final Tuple2<Integer, Row> tup2 : indexedRows.collect(Collectors.toList())) {
+                final int rowIdx = tup2._1();
+                final Row row    = tup2._2();
+
+                if (changedRowIndices.contains(rowIdx)) {
+                    // We only care about the child rows (hence + 1), and all rows have been offset by autoGroupSizeDiff.
+                    final int lowerIdx = tableRowIdx + 1 + autoGroupSizeDiff;
+                    final int upperIdx = lowerIdx + row.children().size() - 1;
+                    fireTableRowsUpdated(lowerIdx, upperIdx);
+                }
```

Comment:
The only thing that is changing in this case is the auto group, correct?  The indices of other rows might be adjusted, but does that require a row update event?  I'm likely overlooking the obvious but the auto group insertion/deletions should be sufficient I believe.  Also I would have guessed that firing update events for the other rows that haven't changed would defeat the purpose of trying to limit the event to the minimum possible change.

Did you test this with GeMS by the way?  It is maybe the only current AGS strategy where multiple auto group rows might be added at once.